core/src/test/java/org/kohsuke/stapler/export/JSONDataWriterTest.java

```diff
@@ -74,4 +78,48 @@ public Supers(Super... elements) {
         }
     }
 
+    @Test
+    public void JSONObjectWithoutNullValue() throws IOException {
+        String json = "{'key':'value', 'optional':'present'}";
+
+        StringWriter w = new StringWriter();
+        ModelWithJsonField jsonModel = new ModelWithJsonField(json);
+
+        DataWriter writer = JSON.createDataWriter(jsonModel,w);
+        Model<ModelWithJsonField> model = new ModelBuilder().get(ModelWithJsonField.class);
+        model.writeTo(jsonModel, writer);
+
+        assertTrue("Generated JSON :"+w.toString()+ "is not a correct representation of :"+json,
+                   w.toString().contains("present"));
+    }
+
+    @Test
+    public void JSONObjectWithNullValue() throws IOException {
+        String json = "{'key':'value', 'optional':null}";
+
+        StringWriter w = new StringWriter();
+        ModelWithJsonField jsonModel = new ModelWithJsonField(json);
+
+        DataWriter writer = JSON.createDataWriter(jsonModel,w);
+        Model<ModelWithJsonField> model = new ModelBuilder().get(ModelWithJsonField.class);
+        model.writeTo(jsonModel, writer);
+
+        assertTrue("Generated JSON :"+w.toString()+ "is not a correct representation of :"+json, w.toString().contains(
+                "null"));
+    }
+
+    @ExportedBean
+    public class ModelWithJsonField {
+        private JSONObject json;
+
+        public ModelWithJsonField(String jsonAsString){
+            json = JSONObject.fromObject(jsonAsString);
```

Comment:
```suggestion
            json = JSONObject.fromObject(jsonAsString.replace('\'', '"');
```

Clearly `json-lib` tolerates invalid input, among many other crazy behaviors.